s3/src/main/java/ch/cyberduck/core/s3/S3WebIdentityTokenExpiredResponseInterceptor.java

```diff
@@ -0,0 +1,126 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+import ch.cyberduck.core.Credentials;
+import ch.cyberduck.core.DisabledCancelCallback;
+import ch.cyberduck.core.Host;
+import ch.cyberduck.core.LoginCallback;
+import ch.cyberduck.core.exception.BackgroundException;
+import ch.cyberduck.core.exception.ExpiredTokenException;
+import ch.cyberduck.core.exception.InteroperabilityException;
+import ch.cyberduck.core.exception.LoginFailureException;
+import ch.cyberduck.core.http.DisabledServiceUnavailableRetryStrategy;
+import ch.cyberduck.core.sts.STSCredentialsRequestInterceptor;
+
+import org.apache.http.HttpResponse;
+import org.apache.http.entity.BufferedHttpEntity;
+import org.apache.http.protocol.HttpContext;
+import org.apache.http.util.EntityUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jets3t.service.S3ServiceException;
+import org.jets3t.service.security.AWSSessionCredentials;
+
+import java.io.IOException;
+
+public class S3WebIdentityTokenExpiredResponseInterceptor extends DisabledServiceUnavailableRetryStrategy {
+    private static final Logger log = LogManager.getLogger(S3WebIdentityTokenExpiredResponseInterceptor.class);
+
+    private static final int MAX_RETRIES = 1;
+
+    private final S3Session session;
+    private final Host host;
+    private final STSCredentialsRequestInterceptor authorizationService;
+    private final LoginCallback prompt;
+
+    public S3WebIdentityTokenExpiredResponseInterceptor(final S3Session session, final LoginCallback prompt,
+                                                        STSCredentialsRequestInterceptor authorizationService) {
+        this.session = session;
+        this.host = session.getHost();
+        this.authorizationService = authorizationService;
+        this.prompt = prompt;
+    }
+
+    @Override
+    public boolean retryRequest(final HttpResponse response, final int executionCount, final HttpContext context) {
+        if(executionCount <= MAX_RETRIES) {
+            if (400 == response.getStatusLine().getStatusCode() || response.getStatusLine().getStatusCode() == 403) {
+                try {
+                    final S3ServiceException failure;
+                    if(null != response.getEntity()) {
+                        EntityUtils.updateEntity(response, new BufferedHttpEntity(response.getEntity()));
+                        failure = new S3ServiceException(response.getStatusLine().getReasonPhrase(),
+                                EntityUtils.toString(response.getEntity()));
+                    }
+                    // minio sometimes packs the error code and description in the http header
+                    else{
+                        failure = new S3ServiceException(response.getStatusLine().getReasonPhrase());
```

Comment:
The InvalidAccessKeyId error code has no response body.
I made a commented [test](https://github.com/SheromS/cyberduck/blob/9eb90698c2c46557fb48139fc4e17266bfbd749b/s3/src/test/java/ch/cyberduck/core/oidc_sts/OidcAuthenticationTest.java#L151C11-L151C11) which shows an edge case scenario. The request leaves the app right before the expiry time is reached. Because of the network latency the expiry time is reach at the minio service. The AccessKey got just outdated and the Failure gets handled in this class `S3WebIdentityTokenExpiredResponseInterceptor`.
What are your thoughts about that?

Reply:
We took your [changes](https://github.com/iterate-ch/cyberduck/pull/14893/files/3528d70556808a5c1c61ad2bd37ec3b5fd0b10dd#r1264642467) with the minio headers. Is that fine for now?